src/BaseTemplates/StarterWeb/Startup.cs

```diff
@@ -19,10 +19,14 @@ public Startup(IHostingEnvironment env, IApplicationEnvironment appEnv)
             // Set up configuration sources.
             var builder = new ConfigurationBuilder()
                 .SetBasePath(appEnv.ApplicationBasePath)
-                .AddJsonFile("appsettings.json")
+                .AddJsonFile(source =>
+                {
+                    source.Path = "appsettings.json";
+                    source.ReloadOnChange = true;
```

Comment:
I now remember there was a big discussion around this and ended up keeping the existing overloads but adding the lambda syntax to support the new parameters (including reload-on-change). Given that I believe we shouldn't try to rush any other changes in and this is the best we can do for RC2 but we should take another look early in the next milestone and if possible cleanup and rationalize the overloads to have something that feels more cohesive. I have filed https://github.com/aspnet/Configuration/issues/420.